src/hotspot/share/opto/loopnode.cpp

```diff
@@ -2274,6 +2275,10 @@ void CountedLoopNode::dump_spec(outputStream *st) const {
 
 //=============================================================================
 jlong BaseCountedLoopEndNode::stride_con() const {
+  if (!stride_is_con()) {
+    // Stride could be non-constant if a loop is vector masked
+    return 0;
```

Comment:
Could this break the assumption anywhere else that `stride_con != 0`?
I fear that it may just silently succeed everywhere, or do checks like:
```
if (stride_con() > 0) {
  // assume positive
} else {
  // assume negative (now wrong!)
}
```
Might it be better to have an assert here, and do the `stride_is_con` checks at the call sites of `stride_con`?

Reply:
I have reverted this change and turned to update `CountedLoopNode::stride_con()` (and add asserts there) to mitigate this potential issue. That one is a call site of this function and "int" counted loop transformation directly calls there. Before my patch, that function may also return 0.